src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,668 @@
+
+                    _writer.WriteBlockStart($@"return {Literal.services}.{Literal.Configure}<{typeDisplayString}>({Literal.obj} =>");
+                    EmitBindLogicFromIConfiguration(type, Literal.obj, InitializationKind.None);
+                    _writer.WriteBlockEnd(");");
+
+                    _writer.WriteBlockEnd();
+                    _writer.WriteBlankLine();
+                }
+
+                Emit_NotSupportedException_UnableToBindType(NotSupportedReason.TypeNotDetectedAsInput);
+                _writer.WriteBlockEnd();
+
+                _writer.WriteBlankLine();
+            }
+
+            private void EmitGetMethod()
+            {
+                if (_generationSpec.TypesForGetMethodGen.Count == 0)
+                {
+                    return;
+                }
+
+                _writer.WriteBlockStart($"public static T? {Literal.Get}<T>(this {GlobalName.IConfiguration} {Literal.configuration})");
+
+                EmitCheckForNullArgument_WithBlankLine(Literal.configuration);
+
+                foreach (TypeSpec type in _generationSpec.TypesForGetMethodGen)
+                {
+                    string typeDisplayString = type.DisplayString;
+
+                    _writer.WriteBlockStart($"if (typeof(T) == typeof({typeDisplayString}))");
+                    EmitBindLogicFromIConfiguration(type, Literal.obj, InitializationKind.Declaration);
+                    _writer.WriteLine($"return (T)({GlobalName.Object}){Literal.obj};");
+                    _writer.WriteBlockEnd();
+                    _writer.WriteBlankLine();
+                }
+
+                Emit_NotSupportedException_UnableToBindType(NotSupportedReason.TypeNotDetectedAsInput);
+                _writer.WriteBlockEnd();
+                _writer.WriteBlankLine();
+            }
+
+            private void EmitBindMethods()
+            {
+                if (_generationSpec.TypesForBindMethodGen.Count > 0)
+                {
+                    foreach (TypeSpec type in _generationSpec.TypesForBindMethodGen)
+                    {
+                        EmitBindMethod(type);
+                    }
+                }
+
+                // Get & Configure method generation might have queued types for private BindCore impl
+                while (_privateBindCoreMethodGen_QueuedTypes.Count > 0)
+                {
+                    EmitBindCoreMethod(_privateBindCoreMethodGen_QueuedTypes.Dequeue());
+                }
+            }
+
+            private void EmitBindMethod(TypeSpec type)
+            {
+                if (_internalBindMethodGen_ProcessedTypes.Contains(type))
+                {
+                    return;
+                }
+
+                _internalBindMethodGen_ProcessedTypes.Add(type);
+
+                // Binding to root level struct is a no-op.
+                // TODO: maybe this should be a debug assert & the parser shouldn't include them.
+                if (type.IsValueType)
```

Comment:
Nit: similarly, this kind of logic should be part of the "parser" for better caching properties. The source formatter should be blindly emitting code for what it was given by your source generator's domain logic.